docs/migrating-to-v4-guide.md

```diff
@@ -0,0 +1,206 @@
+
+_You can find the motivation behind this behavior on [this issue comment](https://github.com/testing-library/eslint-plugin-testing-library/issues/222#issuecomment-679592434)._
+
+### Imports
+
+By default, `eslint-plugin-testing-library` v4 won't check from which module are the utils imported. This means it doesn't matter if you are importing the utils from `@testing-library/*`, `test-utils` or `whatever`.
+
+There is a new Shared Setting to restrict this scope tho: [`utils-module`](#utils-module). By using this setting, only utils imported from `@testing-library/*` packages, or the custom one indicated in this setting would be reported.
+
+### Renders
+
+By default, `eslint-plugin-testing-library` v4 will assume that all methods which names contain "render" should be reported. This means it doesn't matter if you are rendering your elements for testing using `render`, `customRender` or `renderWithRedux`.
+
+There is a new Shared Setting to restrict this scope tho: [`custom-renders`](#custom-renders). By using this setting, only methods strictly named `render` or as one of the indicated Custom Renders would be reported.
+
+### Queries
+
+`eslint-plugin-testing-library` v4 will assume that all methods named following the pattern `get(All)By*`, `query(All)By*`, or `find(All)By*` are queries to be reported. This means it doesn't matter if you are using a built-in query (`getByText`), or a custom one (`getByIcon`): if it matches this pattern, it will be assumed as a potential query to be reported.
+
+There is no way to restrict this behavior for now.
+
+## Shared Settings
+
+ESLint has a setting feature which allows configuring data that must be shared across all its rules: [Shared Settings](https://eslint.org/docs/user-guide/configuring/configuration-files#adding-shared-settings). Since `eslint-plugin-testing-library` v4 we are using this Shared Settings to config global things for the plugin.
+
+To avoid collision with settings from other ESLint plugins, all the properties for this one are prefixed with `testing-library/`.
+
+⚠️ **Please be aware of using these settings will disable part of [Aggressive Reporting](#aggressive-reporting).**
+
+### `utils-module`
+
+Relates to [Aggressive Reporting - Imports](#imports). This setting (just a string) allows you to indicate which is the only Custom Module you'd like to be reported by `eslint-plugin-testing-library`.
+
+```json
+// .eslintrc
+{
+  "settings": {
+    "testing-library/utils-module": "my-custom-test-utils"
```

Comment:
This example makes it seem like this can only be a single string and not an array? Not sure if it'd add too much complexity to support multiple utils modules.

For example a user could have many custom selectors each imported from a different module. Probably not common, but just calling it out.

Reply:
Yes, for now, it's just a single string (it's mentioned few lines above). It shouldn't be that complex to allow multiple modules, but better to address that in a different PR.

Reply:
Issue created to handle this #315